core-services/pdc/src/main/java/org/silverpeas/core/pdc/pdc/service/PdcSearchProcessor.java

```diff
@@ -0,0 +1,139 @@
+
+/**
+ * @author Nicolas Eysseric
+ */
+@Singleton
+@TaxonomySearch
+public class PdcSearchProcessor implements SearchQueryProcessor {
+
+  @Inject
+  private PdcManager pdcManager;
+
+  @Override
+  public List<SearchResult> process(final QueryDescription query,
+      final List<SearchResult> results) {
+
+    SearchContext pdcContext = new SearchContext(query.getSearchingUser());
+
+    List<AxisValueCriterion> axisValueCriteria =
+        AxisValueCriterion.fromFlattenedAxisValues(query.getTaxonomyPosition());
+    axisValueCriteria.forEach(anAxisValueCriterion->pdcContext.addCriteria(anAxisValueCriterion));
+
+    if (!pdcContext.isEmpty()) {
+      // We get silvercontentids according to the search context, author, components and dates
+      try {
+        List<Integer> contentIds = pdcManager
+            .findSilverContentIdByPosition(pdcContext, new ArrayList<>(query.getWhereToSearch()),
+                null, query.getRequestedCreatedAfter(), query.getRequestedCreatedBefore());
+
+        List<GlobalSilverContent> contents = getGlobalSilverContents(contentIds, pdcContext.getUserId());
+        Collections.sort(contents, cDateDesc);
+
+        return toSearchResults(contents);
+      } catch (Exception e) {
+
+      }
+    }
+    return new ArrayList<>();
+  }
+
+  private List<GlobalSilverContent> getGlobalSilverContents(List<Integer> silverContentIds, String userId) throws Exception {
+
+    List<GlobalSilverContent> silverContents = new ArrayList<>();
+    if (silverContentIds == null || silverContentIds.isEmpty()) {
+      return silverContents;
+    }
+
+    ContentManager contentManager = new ContentManager();
+    List<String> instanceIds = contentManager.getInstanceId(silverContentIds);
+
+    for (String instanceId : instanceIds) {
+      // On récupère tous les silverContentId d'un instanceId
+      List<Integer> allSilverContentIds = contentManager.getSilverContentIdByInstanceId(instanceId);
+
+      // une fois les SilverContentId de l'instanceId récupérés, on ne garde que ceux qui sont
+      // dans la liste résultat (alSilverContentIds).
+      allSilverContentIds.retainAll(silverContentIds);
+
+      ContentPeas contentP = contentManager.getContentPeas(instanceId);
+      if (contentP != null) {
+        // we are going to search only SilverContent of this instanceId
+        ContentInterface contentInterface = contentP.getContentInterface();
+        List<SilverContentInterface> silverContentTempo = contentInterface.getSilverContentById(
+            allSilverContentIds, instanceId, userId);
+
+        if (silverContentTempo != null) {
+          silverContents.addAll(transformSilverContentsToGlobalSilverContents(silverContentTempo,
+              instanceId));
+        }
+      }
+    }
+    return silverContents;
+  }
+
+  private List<GlobalSilverContent> transformSilverContentsToGlobalSilverContents(
+      List<SilverContentInterface> silverContentTempo, String instanceId) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00112)